Source/Models/JSONDeserializer.swift

```diff
@@ -0,0 +1,37 @@
+import Argo
+import Result
+
+struct JSONDeserializer: Deserializer {
+  func deserialize<T : Request>(request: T) -> (NSData?) -> Result<T.ResponseParser.Representation, SwishError> {
```

Comment:
So this `request` only exists to act as a hint for what `T` is, right? What if we made `JSONDeserializer` itself generic over `T: ResponseParser` (or would it need to be `Deserializer` that's generic)? I guess that adds the complication that we'd need to move the deserializer instantiation down to the `performRequest` method, which might not be ideal? I don't know. It just always bothers me to have a generic function like this that needs to take an otherwise unused instance of an object to act as a type hint.